yt_dlp/extractor/taptap.py

```diff
@@ -0,0 +1,209 @@
+        video_id = self._match_id(url)
+        query = {self._INFO_QUERY_KEY: video_id}
+
+        data = traverse_obj(
+            self._get_api(self._INFO_API, video_id, query=query), self._DATA_PATH)
+
+        video_ids = traverse_obj(data, self._ID_PATH)
+        metainfo = traverse_obj(data, self._META_PATH)
+        return self._extract_entries(video_ids, metainfo, video_id)
+
+
+class TapTapIntlBase(TapTapBaseIE):
+    _X_UA = 'V=1&PN=WebAppIntl2&LANG=zh_TW&VN_CODE=115&VN=0.1.0&LOC=CN&PLT=PC&DS=Android&UID={uuid}&CURR=&DT=PC&OS=Windows&OSV=NT%208.0.0'
+    _VIDEO_API = 'https://www.taptap.io/webapiv2/video-resource/v1/multi-get'
+
+
+class TapTapMomentIE(TapTapBaseIE):
+    _VALID_URL = r'https?://www\.taptap\.cn/moment/(?P<id>\d+)'
+    _INFO_API = 'https://www.taptap.cn/webapiv2/moment/v3/detail'
+    _ID_PATH = ('moment', 'topic', (('videos', ...), 'pin_video'), 'video_id')
+    _META_PATH = ('moment', {
+        'timestamp': ('created_time', {int_or_none}),
+        'uploader': ('author', 'user', 'name', {str}),
+        'title': ('topic', 'title', {str}),
+        'description': ('topic', 'summary', {str}),
+    })
+    _TESTS = [{
+        'url': 'https://www.taptap.cn/moment/194618230982052443',
+        'info_dict': {
+            'id': '194618230982052443',
+            'title': '《崩坏3》开放世界「后崩坏书」新篇章 于淹没之地仰视辰星',
+            'description': 'md5:cf66f7819d413641b8b28c8543f4ecda',
+            'timestamp': 1633453402,
+            'upload_date': '20211005',
+            'uploader': '乌酱',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '2202584',
+                'ext': 'mp4',
+                'title': '《崩坏3》开放世界「后崩坏书」新篇章 于淹没之地仰视辰星',
+                'description': 'md5:cf66f7819d413641b8b28c8543f4ecda',
+                'duration': 66,
+                'timestamp': 1633453402,
+                'upload_date': '20211005',
+                'uploader': '乌酱',
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }, {
+        'url': 'https://www.taptap.cn/moment/521630629209573493',
+        'info_dict': {
+            'id': '521630629209573493',
+            'title': '《崩坏：星穹铁道》黄泉角色PV——「你的颜色」',
+            'description': 'md5:2c81245da864428c904d53ae4ad2182b',
+            'timestamp': 1711425600,
+            'upload_date': '20240326',
+            'uploader': '崩坏：星穹铁道',
+        },
+        'playlist_count': 1,
+        'playlist': [{
+            'info_dict': {
+                'id': '4006511',
+                'ext': 'mp4',
+                'title': '《崩坏：星穹铁道》黄泉角色PV——「你的颜色」',
+                'description': 'md5:2c81245da864428c904d53ae4ad2182b',
+                'duration': 173,
+                'timestamp': 1711425600,
+                'upload_date': '20240326',
+                'uploader': '崩坏：星穹铁道',
+                'thumbnail': r're:^https?://.*\.(png|jpg)',
+            }
+        }]
+    }]
```

Comment:
I also went ahead and added another test with more than one video:
```suggestion
        }],
        'params': {'skip_download': 'm3u8'},
    }, {
        'url': 'https://www.taptap.cn/moment/540493587511511299',
        'playlist_count': 2,
        'info_dict': {
            'id': '540493587511511299',
            'title': '中式民俗解谜《纸嫁衣7》、新系列《纸不语》公布！',
            'description': 'md5:d60842350e686ddb242291ddfb8e39c9',
            'timestamp': 1715920200,
            'upload_date': '20240517',
            'modified_timestamp': 1715942225,
            'modified_date': '20240517',
            'uploader': 'TapTap 编辑',
            'uploader_id': '7159244',
        },
        'params': {'skip_download': 'm3u8'},
    }]
```